src/main/java/jenkins/plugins/hipchat/workflow/HipChatStep.java

```diff
@@ -0,0 +1,128 @@
+package jenkins.plugins.hipchat.workflow;
+
+import hudson.AbortException;
+import hudson.Extension;
+import hudson.model.TaskListener;
+import hudson.util.ListBoxModel;
+import jenkins.model.Jenkins;
+import jenkins.plugins.hipchat.HipChatNotifier;
+import jenkins.plugins.hipchat.HipChatService;
+import jenkins.plugins.hipchat.Messages;
+import jenkins.plugins.hipchat.impl.HipChatV1Service;
+import jenkins.plugins.hipchat.impl.HipChatV2Service;
+import jenkins.plugins.hipchat.model.Color;
+import org.apache.commons.lang.StringUtils;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepDescriptorImpl;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepImpl;
+import org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution;
+import org.jenkinsci.plugins.workflow.steps.StepContextParameter;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class HipChatStep extends AbstractStepImpl {
+
+    private static final Logger logger = Logger.getLogger(HipChatStep.class.getName());
+
+    public final String message;
+
+    @DataBoundSetter
+    public String color;
+
+    @DataBoundSetter
+    public String token;
+
+    @DataBoundSetter
+    public String room;
+
+    @DataBoundSetter
+    public String server;
+
+    @DataBoundSetter
+    public boolean notify;
+
+    @DataBoundSetter
+    public Boolean v2enabled;
+
+    @DataBoundSetter
+    public String sendAs;
+
+    @DataBoundConstructor
+    public HipChatStep(@Nonnull String message) {
+        this.message = message;
+    }
+
+    @Extension
+    public static class DescriptorImpl extends AbstractStepDescriptorImpl {
+
+        public DescriptorImpl() {
+            super(HipChatStepExecution.class);
+        }
+
+        @Override
+        public String getFunctionName() {
+            return "hipchat";
+        }
+
+        @Override
+        public String getDisplayName() {
+            return "Publish HipChat Message";
+        }
+
+        public ListBoxModel doFillColorItems(){
```

Comment:
Is this an unused method? I'm not really familiar with all the hidden extension points one can have with jelly, but maybe config.jelly should have had f:enum instead of f:entry?